app/models/school.rb

```diff
@@ -116,6 +116,8 @@ class School < ApplicationRecord
   enum school_type_group: SCHOOL_TYPE_GROUPS
   enum school_type: SCHOOL_TYPES
 
+  has_one :eligible_fe_provider, foreign_key: :ukprn, primary_key: :ukprn
```

Comment:
We'll need a database index on the `eligible_fe_providers` table for `ukprn` column only. At the moment it's only a secondary column in another index.

Reply:
it's actually only used in the test for convenience. although now that you've mentioned it and i've given it some thought i think i'll remove this association and code by hand in the tests. i don't think we need the index as we should never look up just by `ukprn` and should always in tandem with `academic_year`

Reply:
no, actually there is bug as it is actually used in the code and will fix

Reply:
this is now an instance method and defaults to use `AcademicYear.current`, which should be fine for now. If there needs to be any processing in another academic year we may want to store the academic year as part of their answers when they start their journey